psi4/src/psi4/libmints/molecule.cc

```diff
@@ -2221,47 +2221,100 @@ std::vector<std::string> Molecule::irrep_labels() {
+double Molecule::y(int atom) const {
+    if (atom >= atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return input_units_to_au_ * atoms_[atom]->compute()[1];
+}
 
-double Molecule::z(int atom) const { return input_units_to_au_ * atoms_[atom]->compute()[2]; }
+double Molecule::z(int atom) const {
+    if (atom >= atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return input_units_to_au_ * atoms_[atom]->compute()[2];
+}
 
-double Molecule::fx(int atom) const { return input_units_to_au_ * full_atoms_[atom]->compute()[0]; }
+double Molecule::fx(int atom) const {
+    if (atom >= full_atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return input_units_to_au_ * full_atoms_[atom]->compute()[0];
+}
 
-double Molecule::fy(int atom) const { return input_units_to_au_ * full_atoms_[atom]->compute()[1]; }
+double Molecule::fy(int atom) const {
+    if (atom >= full_atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return input_units_to_au_ * full_atoms_[atom]->compute()[1]; 
+}
 
-double Molecule::fz(int atom) const { return input_units_to_au_ * full_atoms_[atom]->compute()[2]; }
+double Molecule::fz(int atom) const {
+    if (atom >= full_atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return input_units_to_au_ * full_atoms_[atom]->compute()[2];
+}
 
-double Molecule::charge(int atom) const { return atoms_[atom]->charge(); }
+double Molecule::charge(int atom) const {
+    if (atom >= atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return atoms_[atom]->charge();
+}
 
-double Molecule::fcharge(int atom) const { return full_atoms_[atom]->charge(); }
+double Molecule::fcharge(int atom) const {
+    if (atom >= full_atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return full_atoms_[atom]->charge();
+}
 
-int Molecule::mass_number(int atom) const { return atoms_[atom]->A(); }
+int Molecule::mass_number(int atom) const {
+    if (atom >= atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return atoms_[atom]->A();
+}
 
-int Molecule::fmass_number(int atom) const { return full_atoms_[atom]->A(); }
+int Molecule::fmass_number(int atom) const {
+    if (atom >= full_atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    return full_atoms_[atom]->A();
+}
 
-void Molecule::set_nuclear_charge(int atom, double newZ) { atoms_[atom]->set_nuclear_charge(newZ); }
+void Molecule::set_nuclear_charge(int atom, double newZ) {
+    if (atom >= atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
+    atoms_[atom]->set_nuclear_charge(newZ);
+}
 
-const std::string &Molecule::basis_on_atom(int atom) const { return atoms_[atom]->basisset(); }
+const std::string &Molecule::basis_on_atom(int atom) const {
+    if (atom >= atoms_.size()) throw std::runtime_error("Requested atom doesn't exist.");
```

Comment:
Would it make more sense if this was a utility function `void check_atom(int atom)`?

Might make things slightly simpler.